custom_components/tuya_local/devices/royalclima_fresh_climate.yaml

```diff
@@ -0,0 +1,409 @@
+name: Air conditioner
+products:
+  - id: mcnywzjlugvvvnjj
+    name: Royal fresh full inverter
+primary_entity:
+  entity: climate
+  translation_key: aircon_extra
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: cold
+              value: cool
+            - dps_val: hot
+              value: heat
+            - dps_val: wet
+              value: dry
+            - dps_val: wind
+              value: fan_only
+            - dps_val: auto
+              value: heat_cool
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 160
+        max: 880
+      mapping:
+        - scale: 10
+          step: 5
+      unit: C
+    - id: 3
+      name: current_temperature
+      type: integer
+      unit: C
+    - id: 4
+      type: string
+      name: mode
+      hidden: true
+    - id: 5
+      type: string
+      name: fan_mode
+      mapping:
+        - dps_val: auto
+          value: auto
+        - dps_val: mute
+          value: quiet
+        - dps_val: low
+          value: low
+        - dps_val: mid_low
+          value: medlow
+        - dps_val: mid
+          value: medium
+        - dps_val: mid_high
+          value: medhigh
+        - dps_val: high
+          value: high
+        - dps_val: strong
+          value: strong
+    - id: 102
+      name: preset_mode
+      type: string
+      translation_key: Recirculate
```

Comment:
translation_key can only be specified at entity level, and also needs to be lower case.
Perhaps some preset_mode mappings can be added to aircon_extra to cover these, or separate preset_mode into a fan entity with class ventilation, mapping these to "circulate", 2 of "eco"/"sleep"/"anti-condensation", "fresh" and "auto" (use dp 1 as the switch for the fan as well) since this isn't really a preset of the climate control, but a separate control for air circulation.